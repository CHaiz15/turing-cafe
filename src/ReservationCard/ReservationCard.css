ul {
  list-style-type: none;
}

.res-card {
  display: flex;
  flex-direction: column;
  align-items: center;
  text-align: center;
  border: solid #028080 4px;
  border-radius: 5%;
  background-color: white;
  height: 20%;
  width: 20%;
  margin: 1%;
  color: #028080;
}

.res-btn {
  background-color: #028080;
  color: white;
  width: 50%;
  margin: 0 0 1% 0;
}