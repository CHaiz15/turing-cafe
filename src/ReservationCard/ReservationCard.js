import React, { Component } from 'react'
import './ReservationCard.css'

class ReservationCard extends Component {
  constructor() {
    super();

    this.state = {

    }
  }

  render() {
    return (
      <div className='res-card'>
        <h3>{this.props.name}</h3>
          <p>{this.props.date}</p>
          <p>{this.props.time}</p>
          <p>Number of guests: {this.props.number}</p>
        <button className='res-btn' onClick={() => this.props.deleteReservation(this.props.id)}>Cancel</button>
      </div>
    )
  }
}

export default ReservationCard