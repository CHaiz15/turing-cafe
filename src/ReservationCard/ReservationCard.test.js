import React from 'react';
import ReservationCard from './ReservationCard'
import { shallow } from 'enzyme'

it('should be an instance of an ReservationCard componenet', () => {
  let wrapper = shallow(
  <ReservationCard 
    id={1}
    name='Caleb'
    date='02/12'
    time='7:00'
    number={15}
    deleteReservation={jest.fn()}
  />);
  expect(wrapper).toMatchSnapshot();
})

it('should call addReservation when button is clicked', () => {
  const deleteReservationMock = jest.fn();
  const wrapper = shallow(
    <ReservationCard 
    id={1}
    name='Caleb'
    date='02/12'
    time='7:00'
    number={15}
    deleteReservation={deleteReservationMock}
  />);
  wrapper.find('button').simulate('click')
  expect(deleteReservationMock).toHaveBeenCalled();
})