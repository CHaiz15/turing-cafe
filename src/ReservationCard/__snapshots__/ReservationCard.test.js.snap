// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be an instance of an ReservationCard componenet 1`] = `
<div
  className="res-card"
>
  <h3>
    Caleb
  </h3>
  <p>
    02/12
  </p>
  <p>
    7:00
  </p>
  <p>
    Number of guests: 
    15
  </p>
  <button
    className="res-btn"
    onClick={[Function]}
  >
    Cancel
  </button>
</div>
`;
